### DISCUSSION TITLE
Virtualizing WrapPanel?

### DISCUSSION THREAD

**Discussion by Shivansps** [AvaloniaUI/Avalonia#13422] [Q&A]
Something i noticed with the upgrade to Avalonia v11 is that there is not no way to virtualize something like this
![image](https://github.com/AvaloniaUI/Avalonia/assets/2444336/4f46a752-d6bc-4093-8dfd-b03e999ef8f6)

With the removal of ItemsRepeater and the virtualise property of ItemsControl, i just cant do it, because i need to use a wrappanel as the  panel template, as the VirtualizingStackPanel is not going to work for this.

**Comment by thevortexcloud** [comment]
There are various third party visualising wrap panel controls. But they mostly have limitations (EG all items must be the same width/height). The most recent one I saw was this: 

https://github.com/AvaloniaUI/Avalonia/pull/11251

Which was not merged for the limitations it has.

**Comment by msneijders** [comment]
Just a thought, but maybe using a VirtualizingStackPanel in which an item is a row of, in this case, 11 child items? recalculate the items in a row when its parents width changes.

But ItemsRepeater is not removed? It has moved to a separate nuget package `Avalonia.Controls.ItemsRepeater`.

**Reply by timunie** [reply]
ItemsRepeater got it's own package in 11.0 just like DataGrid as it is less used than other ItemsControls. Also style needs to be included just like DataGrid. 

If you think it's missing in the docs, a PR to improve the docs are welcome 🤗
